### ISSUE TITLE
Additional papers to cite about linear attraction error made by Wu et al 2023/4 baseline Transformer: consistent with Petty and Frank 2021 "Transformers Generalize Linearly" ( https://arxiv.org/pdf/2109.12036 )

### ISSUE THREAD

**Issue by willy-b** [willy-b/RASP-for-ReCOGS#43]
We should add a note that the linear attraction errors predicted by us (this RASP-for-ReCOGS paper) from observations while constructing our RASP model by hand and confirmed to be made by the baseline Wu et al 2023/4 on the task (see Figure below, Figure 7 in the paper), are consistent with findings in Petty and Frank 2021 "Transformers Generalize Linearly" ( https://arxiv.org/pdf/2109.12036 ) (a paper I found researching prior work on this).

<img src="https://github.com/willy-b/RASP-for-ReCOGS/blob/main/attraction_error_example.png?raw=true"/>

(Figure 7 from our paper https://raw.githubusercontent.com/willy-b/RASP-for-ReCOGS/main/rasp-for-recogs_pos-wbruns-2024-draft.pdf is given above)

**Comment by willy-b** [comment]
Also "Grokking of Hierarchical Structure in Vanilla Transformers" https://arxiv.org/abs/2305.18741 
and "Learning Syntax Without Planting Trees: Understanding Hierarchical Generalization in Transformers" https://direct.mit.edu/tacl/article/doi/10.1162/tacl_a_00733/127877 are relevant papers. Though note we show with our RASP model in this paper that there are non-hierarchical rules (masking out flat patterns before processing) that can avoid the "attraction" errors made by the naive non-hierarchical model here in the COGS obj-pp-to-subj-pp generalization split, that we believe could be learned outside that split, from training data.
